src/marks/auto.js

```diff
@@ -100,9 +100,9 @@ export function auto(data, {x, y, color, size, fx, fy, mark} = {}) {
         : xZero || yZero || colorReduce != null // histogram or heatmap
         ? "bar"
         : x && y
-        ? isContinuous(x) && isContinuous(y) && (xReduce != null || yReduce != null || isMonotonic(x) || isMonotonic(y))
-          ? "line"
-          : "dot"
+        ? isOrdinal(x) || isOrdinal(y) || (xReduce == null && yReduce == null && !isMonotonic(x) && !isMonotonic(y))
+          ? "dot"
+          : "line"
```

Comment:
(This is an unrelated change where I just wanted to get rid of isContinuous by replacing it with !isOrdinal. Sorry for the churn but hopefully it’s clear that this is not changing behavior!)